### PR TITLE
[Diagnostics.AspNet]: Adds local tests for Trace.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
Error reporting is already fairly covered by testing on OWIN.

Towards #4658

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.ClientApp/Controllers/TraceController.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            string message = EntryData.GetMessage(nameof(TraceStackTrace), id);
+            using (CloudTrace.Tracer.StartSpan(message))
+            {
+                Thread.Sleep(10);
+                CloudTrace.Tracer.SetStackTrace(TraceEntryData.CreateStackTrace());
+            }
+            return message;
+        }
+
+        [HttpGet]
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.ClientApp/Controllers/TraceController.cs:68]: Maybe add summary messages to this and ParentChild, for consistency?

File: apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/Trace/CloudTraceTest.cs [hunk 2]
```diff
@@ -98,14 +99,106 @@ public void Trace_Multiple_Spans()
...
+        // and we currently don't have such a setup.
+        // The application we are using is Google.Cloud.Diagnostics.AspNet.ClientApp and we deploy
+        // on local and test on local.
+        // But for CI all these tests should be skipped. See this comment as to why:
+        // https://github.com/googleapis/google-cloud-dotnet/issues/2167#issuecomment-409605030
+
+        private static readonly Uri PublishedServiceBaseAddress = new Uri("http://localhost:58755");
+
+        [Fact(Skip = "https://github.com/googleapis/google-cloud-dotnet/issues/2167#issuecomment-409605030")]
+        public async Task Trace_Iss()
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/Trace/CloudTraceTest.cs:112]: I don't understand the name of the method here. What is Iss?

File: apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/Trace/CloudTraceTest.cs [hunk 3]
```diff
@@ -98,14 +99,106 @@ public void Trace_Multiple_Spans()
...
+                await httpClient.GetAsync(service);
+            }
+
+            var trace = _polling.GetTrace(service, _startTime);
+
+            TraceEntryVerifiers.AssertParentChildSpan(trace, service, childSpanName);
+            TraceEntryVerifiers.AssertSpanLabelsContains(trace.Spans.First(s => s.Name == childSpanName),
+                new Dictionary<string, string>
+                {
+                        {TraceEntryData.TraceLabel, TraceEntryData.TraceLabelValue }
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/Trace/CloudTraceTest.cs:151]: Nit: odd indentation, and add a space after {

Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/Trace/CloudTraceTest.cs]: Done

**Pull request merged**